src/main/java/hudson/plugins/parameterizedtrigger/BuildInfoExporterAction.java

```diff
@@ -243,40 +380,48 @@ public void buildEnvVars(AbstractBuild<?, ?> build, EnvVars env) {
    * @return
    */
   public Object readResolve() {
-    if (this.lastReference == null) {
-      this.lastReference = new BuildReference(this.buildName, this.buildNumber, Result.NOT_BUILT);
+    if (this.lastReference2 == null) {
+      this.lastReference2 = this.lastReference;
```

Comment:
Again, this is a serialization thing. If lastReference was previously used it will be restored. The problem is that it seems not to be possible to just change lastReference to a more abstract type. At least it didn't worked for me but couldn't explain why it shouldn't.
Overall it probably would be better to use customized serialization (implementing readObject/writeObject) for all that stuff.  One has much more complete control over deserialization even after renaming fields and changing the internals. I will think about how to improve this.